src/topics/troubleshooting.md

```diff
@@ -0,0 +1,63 @@
+# Troubleshooting
+
+In this section you will find ways to troubleshoot when you have problems executing CWL.
+We focus on `cwltool` here but some of these techniques may apply to other CWL Runners.
+
+## Run `cwltool` with `cachedir`
+
+You can use the `--cachedir` option when running a workflow to tell `cwltool` to
+cache intermediate files (files that are not input nor output files, but created
+while your workflow is running). By default, these files are created in a
+temporary directory but writing them to a separate directory makes accessing
+them easier.
+
+In the following example `troubleshooting-wf1.cwl` we have two steps, `step_a` and `step_b`.
+The workflow is equivalent to `echo "Hello World" | rev`, which would print the message
+"Hello World" reversed, i.e. "dlroW olleH". However, the second step, `step_b`, **has a typo**,
+where instead of executing the `rev` command it tries to execute `revv`, which
+fails.
+
+```{literalinclude} /_includes/cwl/troubleshooting/troubleshooting-wf1.cwl
+:language: cwl
+:name: "`troubleshooting-wf1.cwl`"
+:caption: "`troubleshooting-wf1.cwl`"
+:emphasize-lines: 42
+```
+
+Let's execute this workflow with `/tmp/cachedir/` as the `--cachedir` value (`cwltool` will
+create the directory for you if it does not exist already):
+
+```{runcmd} cwltool --cachedir /tmp/cachedir/ troubleshooting-wf1.cwl
+:working-directory: src/_includes/cwl/troubleshooting
+:emphasize-lines: 12-14, 19-21
```

Comment:
I don't see any effect of `emphasize-lines` with `{runcmd}`, alas (or at least, not on Read The Docs)

Reply:
You arr correct. I actually had seen it and forgot to log an issue for that.

Runcmd extends the code block directive. It runs a command and then renders the code block.

It looks like the emphasize is applied before the code is rendered. So it gets ignored instead of highlighting the rendered command.

A bit tricky but I guess it could be one of those 1-line code fixes (that alas you spend more than you wanted on it 😅 )